internal/command/quota_model_test.go

```diff
@@ -0,0 +1,300 @@
+			amount:        5,
+			from:          time.Date(2020, 1, 1, 0, 0, 0, 0, time.UTC),
+			resetInterval: time.Hour,
+			limit:         true,
+			notifications: []*quota.SetEventNotification{{
+				ID:      "notification1",
+				Percent: 10,
+				Repeat:  true,
+				CallURL: "https://call.url",
+			}},
+		},
+		args: args{
+			amount:        5,
+			from:          time.Date(2020, 1, 1, 0, 0, 0, 0, time.UTC),
+			resetInterval: time.Hour,
+			limit:         true,
+			notifications: []*QuotaNotification{},
+		},
+		wantEventPayloadJSON: `{"unit":0,"notifications":[]}`,
+		wantErr:              assert.NoError,
+	}, {
+		name: "create new without notification",
+		fields: fields{
+			amount:        5,
+			from:          time.Date(2020, 1, 1, 0, 0, 0, 0, time.UTC),
+			resetInterval: time.Hour,
+			limit:         true,
+			notifications: []*quota.SetEventNotification{{
+				ID:      "notification1",
+				Percent: 10,
+				Repeat:  true,
+				CallURL: "https://call.url",
+			}},
+		},
+		args: args{
+			amount:        5,
+			from:          time.Date(2020, 1, 1, 0, 0, 0, 0, time.UTC),
+			resetInterval: time.Hour,
+			limit:         true,
+			notifications: []*QuotaNotification{},
+		},
+		wantEventPayloadJSON: `{"unit":0,"notifications":[]}`,
+		wantErr:              assert.NoError,
+	}, {
+		name: "create new with all values values",
+		args: args{
+			amount:        5,
+			from:          time.Date(2020, 1, 1, 0, 0, 0, 0, time.UTC),
+			resetInterval: time.Millisecond,
+			limit:         true,
+			notifications: []*QuotaNotification{{
+				Percent: 10,
+				Repeat:  true,
+				CallURL: "https://call.url",
+			}},
+			idGenerator: id_mock.NewIDGeneratorExpectIDs(t, "notification1"),
+			createNew:   true,
+		},
+		wantEventPayloadJSON: `{"unit":0,"from":"2020-01-01T00:00:00Z","interval":1000000,"amount":5,"limit":true,"notifications":[{"id":"notification1","percent":10,"repeat":true,"callUrl":"https://call.url"}]}`,
+		wantErr:              assert.NoError,
+	}, {
+		name:                 "create new with zero values",
+		args:                 args{createNew: true},
+		wantEventPayloadJSON: `{"unit":0,"from":"0001-01-01T00:00:00Z","interval":0,"amount":0,"limit":false,"notifications":[]}`,
+		wantErr:              assert.NoError,
+	}}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			wm := &quotaWriteModel{
+				from:          tt.fields.from,
+				resetInterval: tt.fields.resetInterval,
+				amount:        tt.fields.amount,
+				limit:         tt.fields.limit,
+				notifications: tt.fields.notifications,
+			}
+			gotChanges, err := wm.NewChanges(tt.args.idGenerator, tt.args.createNew, tt.args.amount, tt.args.from, tt.args.resetInterval, tt.args.limit, tt.args.notifications...)
+			if !tt.wantErr(t, err, fmt.Sprintf("NewChanges(%v, %v, %v, %v, %v, %v)", tt.args.createNew, tt.args.amount, tt.args.from, tt.args.resetInterval, tt.args.limit, tt.args.notifications)) {
+				return
+			}
+			bytes, err := json.Marshal(quota.NewSetEvent(
```

Comment:
Why do you marshall the event into a JSON and define the expected also in a JSON? You could just define everything as an event and test the unmarshalling so that the attributes are defined with omitempty?

Reply:
done https://github.com/zitadel/zitadel/pull/6597/commits/1966a00266179ebd3cbdac97d9991529d9d1aab7